packages/patternfly-react/src/components/DualListSelector/DualList.js

```diff
@@ -0,0 +1,189 @@
+      value
+    );
+
+    this.setState({
+      [side]: { ...this.state[side], filterTerm: value, filterResults }
+    });
+  };
+
+  onMainCheckboxChange = e => {
+    const { checked, attributes } = e.target;
+    const side = attributes.side.value;
+    const itemsCopy = this.state[side].items.map(item => {
+      item.checked = checked;
+      return item;
+    });
+    this.setState({
+      [side]: {
+        ...this.state[side],
+        items: itemsCopy,
+        selectCount: checked ? itemsCopy.length : 0
+      }
+    });
+  };
+
+  getFilteredResults = (array, value) =>
+    array.filter(
+      item => item.label.toLowerCase().indexOf(value.toLowerCase()) > -1
+    );
+
+  moveTo = to => {
+    const from = to === 'right' ? 'left' : 'right';
+    const toState = this.state[to];
+    const fromState = this.state[from];
+
+    let toList = [...toState.items];
+    let fromList = [...fromState.items].filter(item => {
+      if (item.checked) {
+        toList.push(item);
+        return false;
+      }
+      return true;
+    });
+
+    toList = this.resetArray(toList, to);
+    fromList = this.resetArray(fromList, from);
+
+    const toFilteredList = toState.filterTerm
+      ? this.getFilteredResults(toList, toState.filterTerm)
+      : null;
+
+    const fromFilteredList = fromState.filterTerm
+      ? this.getFilteredResults(fromList, fromState.filterTerm)
+      : null;
+
+    this.setState({
+      [from]: {
+        ...fromState,
+        items: fromList,
+        selectCount: 0,
+        filterResults: fromFilteredList
+      },
+      [to]: {
+        ...toState,
+        items: toList,
+        selectCount: fromState.selectCount + toState.selectCount,
+        filterResults: toFilteredList
+      }
+    });
+  };
+
+  resetArray = (array, side) => {
+    let sortedArray = [...array]
+      .sort((a, b) => (a.label.toLowerCase() > b.label.toLowerCase() ? 1 : -1))
+      .map((item, index) => ({
+        ...item,
+        name: index
+      }));
+
+    if (this.state && this.state[side]) {
+      sortedArray = this.state[side].isSortAsc
+        ? sortedArray
+        : sortedArray.reverse();
+    }
+
+    return sortedArray;
+  };
+
+  render() {
+    return (
+      <div className="pf-dual-list">
+        <DualListSelector
+          {...this.state.left}
+          onItemChange={this.onItemChange}
+          onSortClick={this.onSortClick}
+          onFilterChange={this.onFilterChange}
+          onMainCheckboxChange={this.onMainCheckboxChange}
+        />
+        <DualListArrows
+          leftArrowClick={e => this.moveTo('left')}
+          rightArrowClick={e => this.moveTo('right')}
+        />
+        <DualListSelector
+          {...this.state.right}
+          onItemChange={this.onItemChange}
+          onSortClick={this.onSortClick}
+          onFilterChange={this.onFilterChange}
+          onMainCheckboxChange={this.onMainCheckboxChange}
+        />
+      </div>
+    );
+  }
+}
+
+DualList.propTypes = {
```

Comment:
It looks like currently, you can't do anything with the list values? It is all contained in local state.  This needs to either be controlled by having the consumer manage the lists through callbacks and values or an onChange needs to be available. Controlled is a more work, but usually gives more freedom to the consumer.